bin/analyze

```diff
@@ -0,0 +1,367 @@
+
+    parser.add_argument("-d", "--debug",
+                        action='store_true')
+
+    parser.add_argument("-w", "--write-to",
+                        default='results_dir',
+                        help="Path for output results")
+    return parser.parse_args()
+
+
+def run_metrics(items, categories, date_range, is_code, conds, period):
+    """
+    Calculate values of metrics on the given data (items).
+
+    :param items: A list of dictionaries.
+        Each item is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+
+    :param date_range: A tuple which represents the period of interest
+        It is of the form (since, until), where since and until are strings.
+        Either, or both can be None. If, for example, since is None, that
+        would mean that all commits from the first commit to the commit
+        who last falls inside the until range will be included.
+
+    :param is_code:  list of Code Condition objects
+        Used to determine what comprises source code.
+
+    :param conds: list of Commit sub-class objects.
+        Used to add restrictions on which commits are
+        included in the analysis.
+
+    :param period: A string which can be any one of the pandas time
+        series rules:
+            'W': week
+            'M': month
+            'D': day
+
+    :returns results: A dictionary with the computed values of metrics,
+        groups according to the category of metrics.
+    """
+
+    logging.info("Running metrics")
+    results = {
+        COMMIT_CATEGORY: [],
+        ISSUE_CATEGORY: [],
+        PULL_REQUEST_CATEGORY: []
+    }
+
+    for category in categories:
+        for metric in METRICS[category]:
+
+            if category == COMMIT_CATEGORY:
+                metric_obj = metric(items[category], date_range,
+                                    is_code, conds)
+            else:
+                metric_obj = metric(items[category], date_range)
+
+            logging.info("Computing the value of: %s" % str(metric_obj))
+
+            result = {
+                        'metric': metric_obj,
+                        'value': metric_obj.compute(),
+                    }
+
+            results[category].append(result)
+
+    logging.info("Metrics computed")
+
+    return results
+
+
+def fetch_data(owner, repository, api_token, categories):
+    """
+    Fetches data required for the analysis.
+
+    Depending on the category of metrics to be computed, this function
+    calls the Git (commits) and the GitHub backend (issues, pull_requests).
+
+    :param owner: The owner of a repository
+    :param repository: The name of a git repository
+    :param api_token: GitHub API token
+    :categories: A list of metric categories. A combination of
+        COMMIT_CATEGORY, ISSUE_CATEGORY and PULL_REQUEST_CATEGORY.
+
+    :returns data: A dictionary of fetched items, segregated by category.
+    """
+
+    if api_token is None:
```

Comment:
minor comment, this if statement could be rewritten in a more pythonic way:
`api_token = [] if api_token is None else api_token`